plugin/pkg/admission/broker/authsarcheck/admission.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authsarcheck
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	kubeinformers "k8s.io/client-go/informers"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	authorizationapi "k8s.io/client-go/pkg/apis/authorization/v1"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	pluginName = "BrokerAuthSarCheck"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(pluginName, func(io.Reader) (admission.Interface, error) {
+		return NewSARCheck()
+	})
+}
+
+// sarcheck is an implementation of admission.Interface.
+// It enforces the creator of a broker has proper access to the auth credentials
+type sarcheck struct {
+	*admission.Handler
+	client kubeclientset.Interface
+}
+
+var _ = scadmission.WantsKubeInformerFactory(&sarcheck{})
+var _ = scadmission.WantsKubeClientSet(&sarcheck{})
+
+func convertToSARExtra(extra map[string][]string) map[string]authorizationapi.ExtraValue {
+	if extra == nil {
+		return nil
+	}
+
+	ret := map[string]authorizationapi.ExtraValue{}
+	for k, v := range extra {
+		ret[k] = authorizationapi.ExtraValue(v)
+	}
+
+	return ret
+}
+
+func (s *sarcheck) Admit(a admission.Attributes) error {
+	// need to wait for our caches to warm
+	if !s.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+	// only care about brokers
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("servicebrokers") {
+		return nil
+	}
+	broker, ok := a.GetObject().(*servicecatalog.ServiceBroker)
+	if !ok {
+		return errors.NewBadRequest("Resource was marked with kind Broker but was unable to be converted")
+	}
+	glog.V(5).Infof("Retrieved broker %v", broker)
+
+	if broker.Spec.AuthInfo == nil {
+		// no auth secret to check
+		return admission.NewForbidden(a, fmt.Errorf("broker does not have authentication information"))
+	}
+
+	var secretRef *v1.ObjectReference
+	if broker.Spec.AuthInfo.Basic != nil {
```

Comment:
handle the case when SecretRef is not provided (no auth is configured in the broker, or the authInfo is invalid, missing secretRef)?
i.e. when secretRef is `nil`.